mw-webapp/src/component/editableText/EditableText.tsx

```diff
@@ -69,24 +70,23 @@ export const EditableText = <T extends string | number>(props: EditableTextProps
   };
 
   /**
-   * Check type of coming value and convert it to Number if need to use input with type "number"
+   * Update value
    */
-  const setValue = (value: string) => {
-    const number = Number(value) ?? 0;
-    const updatedValue = props.type === "number" ? number : value;
-    setText(updatedValue as T);
+  const updateValue = (updatedValue: string | number) => {
+    setValue(updatedValue as T);
   };
 
   /**
    * Render input
    */
   const renderInput = () => (
     <Input
+      formatter={FormatterInputValue.withNoFirstZero}
       type={props.type ?? "text"}
       max={props.max}
-      value={text}
+      value={value}
```

Comment:
use type guard for separating two different cases for formater string and number

Reply:
Used